src/Editor.elm

```diff
@@ -434,6 +396,50 @@ viewNodeInputs additional z =
         )
 
 
+ruleMenu unaryMsg unaryWithSubstMsg binaryMsg label labelPrefix cls config z =
+    let
+        item rule itemToShow =
+            if Dict.get rule config |> Maybe.withDefault False then
+                itemToShow
+
+            else
+                text ""
+
+        unaryItem extType =
+            item (unaryExtTypeToString extType) <|
+                menuItem (unaryMsg extType z) (labelPrefix ++ " " ++ unaryExtTypeToString extType)
+
+        unaryWithSubstItem extType =
+            item (unaryWithSubstExtTypeToString extType) <|
+                menuItem (unaryWithSubstMsg extType z) (labelPrefix ++ " " ++ unaryWithSubstExtTypeToString extType)
+
+        binaryItem extType =
+            item (binaryExtTypeToString extType) <|
+                menuItem (binaryMsg extType z) (labelPrefix ++ " " ++ binaryExtTypeToString extType)
+    in
+    menu cls label <|
+        [ unaryItem Alpha
+        , binaryItem Beta
+        ]
+            ++ List.map unaryWithSubstItem [ Gamma, Delta, GammaStar, DeltaStar ]
+            ++ List.map unaryItem [ Refl, Leibnitz, MP, MT, HS, DS, NCS, ESFF, ESFT, ESTF, ESTT ]
+            ++ List.map binaryItem [ Cut, ECDF, ECDT ]
+
+
+menuItem : Msg -> String -> Html Msg
+menuItem msg str =
+    li [] [ button [ onClick msg ] [ text str ] ]
+
+
+menu cls label content =
+    div [ class "onclick-menu" ]
+        [ button [ class <| "onclick-menu " ++ cls, tabindex 0 ]
+            [ text label
+            , ul [ class "onclick-menu-content" ] <| content
+            ]
+        ]
```

Comment:
Maybe there was a misunderstanding – the outer `div` is pointless. The inner `button` with `class` and `tabindex` must not be a `button` but a `div`. Otherwise it doesn’t work in macOS (except in Chrome).